src/toolbars/TimeToolBar.cpp

```diff
@@ -293,7 +293,7 @@ void TimeToolBar::OnUpdate(wxCommandEvent &evt)
 
    // Save format name before recreating the controls so they resize properly
    if (mListener) {
-      mListener->TT_SetAudioTimeFormat(mAudioTime->GetBuiltinName(evt.GetInt()));
+      mListener->TT_SetAudioTimeFormat(evt.GetString());
```

Comment:
Oh no -- now I think you should do all the work of using Observer, after all.

Whereas `TT_SetAudioTimeFormat` received a `NumericFormatSymbol` before, now instead it gets `NumericFormatSymbol` constructed from the internal identifier only, possibly losing translatable string information.

In fact, in the tables now, all the NumericFormatSymbols are like `{ XO("seconds") }`,
that is using the constructor `ComponentInterfaceSymbol( const TranslatableString &msgid )`, so none will really lose information.

Yet, it doesn't work now with the correct, future-proof generality.

Reply:
Similar remarks about the calls to `SSBL_SetFrequencySelectionFormatName`, `SSBL_SetBandwidthSelectionFormatName`

Reply:
And yet again -- I do have that other pending branch to clean up all that stuff, eliminating those bogus listener classes.

Maybe it would be less work for us all if I simply pushed that bit of work onto your brnach!  That would avoid the problem.

Reply:
I was under impression that the match is done by `Internal`. I don't really understand what could be the reason behind the locale specific matching for identifiers.

If this is the case - saving to the configuration was broken long before this PR.

Reply:
It's not saving to config that would be broken.  I just haven't followed the chain of consequences to see whether some l10n would be broken afterward in-session.

And even then it might only become broken if for any reason the tables had to change their English strings and use the two-argument constructor of the symbol to keep compatibiliyt (but that is unlikely to happen).  So it would be not broken, but fragile.

But as I say, really the selection bar listener classes should be eliminated, and I have a commit already to do that which does not require all else that is in my branch.  I would like to append that commit to this branch.

Reply:
In any case, you have not convinced me that I am wrong. 

Citing ComponentInterfaceSymbol implementation:

```
   friend inline bool operator == (
      const ComponentInterfaceSymbol &a, const ComponentInterfaceSymbol &b )
   { return a.mInternal == b.mInternal; }

   friend inline bool operator != (
      const ComponentInterfaceSymbol &a, const ComponentInterfaceSymbol &b )
   { return !( a == b ); }

   //! Comparator for use in ordered containers, using internal name only
   friend inline bool operator < (
      const ComponentInterfaceSymbol &a, const ComponentInterfaceSymbol &b )
   { return a.mInternal < b.mInternal; }
```

We do store `Internal` in preferences, so I don't think that I have introduced any kind of bug here.

Reply:
There is more in that class than just its equality and comparison operators.

Reply:
There is. But we only use equality here and we store this value in preferences. And this value **was** stored in preferences before. So if it will break - it will break globally, not because code here relies only on Internal part.

For display purposes  NumericFormatSymbol can still be retrieved from the registry. This is how the menu is built!

Reply:
Allow me to push the commit that removes the listener classes so I can relieve my doubts, the code gets simpler, and we can stop arguing about this point.

Reply:
Sure, go ahead